pkg/apis/security/validation/tokenrequest.go

```diff
@@ -0,0 +1,67 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package validation
+
+import (
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/gardener/gardener/pkg/apis/security"
+)
+
+// ValidateTokenRequest validates a TokenRequest.
+func ValidateTokenRequest(request *security.TokenRequest) field.ErrorList {
+	allErrs := field.ErrorList{}
+	var (
+		specPath = field.NewPath("spec")
+	)
+	const (
+		minDuration = time.Minute * 10 // TODO(vpnachev): Make min duration configurable.
+		maxDuration = time.Hour * 48   // TODO(vpnachev): Make max duration configurable.
+	)
+	if request.Spec.Duration != nil {
+		if request.Spec.Duration.Duration < minDuration {
+			allErrs = append(allErrs, field.Invalid(specPath.Child("duration"), request.Spec.Duration.String(), "may not specify a duration shorter than "+minDuration.String()))
+		}
+		if request.Spec.Duration.Duration > maxDuration {
+			allErrs = append(allErrs, field.Invalid(specPath.Child("duration"), request.Spec.Duration.String(), "may not specify a duration longer than "+maxDuration.String()))
+		}
+	}
```

Comment:
Shouldn't we just adjust the token expiration in the strategy if the spec does not correspond to the configured min/max similar to the `AdminKubeconfig` and `serviceaccount/tokenrequest` APIs?

Reply:
I think here a similar approach is followed. 
`AdminKubeconfig` has static validation which ensures the `expirationSeconds` is within certain limits (those are static, non-configurable limits that are more like sanity check for the set value)  https://github.com/gardener/gardener/blob/3de8d8b86c8c72a7812f4bf2960d794ed09d5336/pkg/apis/authentication/validation/validation.go#L24-L29 and then in the storage the `expirationSeconds` is capped by the configurable `maxExpirationSeconds` https://github.com/gardener/gardener/blob/3de8d8b86c8c72a7812f4bf2960d794ed09d5336/pkg/apiserver/registry/core/shoot/storage/kubeconfig.go#L144-L146


What is different in the `TokenRequest`, there is no static defaulting. The reason is that the GEP envision the min, max and default duration to be configurable, and changing default values is considered backward incompatible change, therefore there is no explicit defaulting. 

But generally, you are right - having a static validation and mechanism that both ensures the token duration is within the same configurable limits sounds odd. 

K8s seems to be doing just like Gardener for AdminKubeconfig
- [static defaulting](https://github.com/kubernetes/kubernetes/blob/1ebc3d2a6480f3679c28d0bc3d486ad93e95084f/pkg/apis/authentication/v1/defaults.go#L30-L31)
- [static validation, doing sanity check for the value of the field](https://github.com/kubernetes/kubernetes/blob/1ebc3d2a6480f3679c28d0bc3d486ad93e95084f/pkg/apis/authentication/validation/validation.go#L33-L39) 
- [caps the duration with configurable maxDuration](https://github.com/kubernetes/kubernetes/blob/1ebc3d2a6480f3679c28d0bc3d486ad93e95084f/pkg/registry/core/serviceaccount/storage/token.go#L208) 


For simplicity, let's apply the same for the Workload Identity tokens. I will adapt the GEP to drop the configurable default duration.